src/common/models/utils.ts

```diff
@@ -0,0 +1,54 @@
+import _ from 'lodash';
+
+// TODO(michael): We should perhaps create a Timeseries class that wraps an
+// Array<Date> and Array<number | null> and hang these utility functions off of
+// that.
+
+export function indexOfLastValue<T>(data: Array<T | null>): number | null {
+  for (let i = data.length - 1; i >= 0; i--) {
+    if (data[i] !== null && data[i] !== undefined) {
+      return i;
+    }
+  }
+  return null;
+}
+
+export function lastValue<T>(data: Array<T | null>): T | null {
+  const i = indexOfLastValue(data);
+  return i === null ? null : data[i]!;
+}
+
+/**
+ * Replaces all values after the specified date with nulls.
+ */
+export function omitDataAfterDate<T>(
+  data: Array<T | null> | Array<T>,
+  dates: Array<Date>,
+  cutoffDate: Date,
+): Array<T | null> {
+  return dates.map((date, idx) => {
+    const value = data[idx];
+    return date <= cutoffDate ? value : null;
+  });
+}
+
+/**
+ * Returns true if the series has at least 7 non-null values in the last 14
+ * days and has ever had a non-zero value.
```

Comment:
This is an arbitrary criteria I came up with for when actuals are good enough to show.  Seems okay for now.